### ISSUE TITLE
Ignores all errors when using `noqa: X` to only ignore X

### ISSUE THREAD

**Issue by blueyed** [PyCQA/flake8-bugbear#44]
When using `# noqa: E501` all errors are excluded.

This is due to the code in https://github.com/PyCQA/flake8-bugbear/blob/6bf40eafe99bb569296c460b691a6c0bebd27ea5/bugbear.py#L39-L40.

`pycodestyle.noqa` searches for r'# no(?:qa|pep8)\b', which matches `# noqa: E501` also.

Without knowing the internals it seems like flake8 itself should include the errors itself - it has logic to ignore only specific errors (https://github.com/PyCQA/flake8/blob/013b925624bd5e4c026e83eb2c6e9f45cafa0dc0/src/flake8/defaults.py#L39-L51).

Example:
```python
s = 'some very, very, very, very, very, very, very, very, very, very, very, very, very, very, very long line'  # noqa: E123
```

flake8-bugbear should report error `t-noqa-colon.py:1:110: B950 line too long (109 > 80 characters)` for it, but is silent.

**Comment by plinss** [comment]
It's considered an [anti-pattern](https://github.com/PyCQA/pydocstyle/issues/484#issuecomment-641263791) for flake8 plugins to filter errors based on `# noqa` comments. Best practice is to simply send all errors and let flake8 filter them based on user settings, e.g. `--disable-noqa` should show these errors, but won't because they never get to flake8.

As an aside, this behavior also breaks a plugin of mine: https://github.com/plinss/flake8-noqa, see https://github.com/plinss/flake8-noqa/issues/1
